spikeinterface/sortingcomponents/peak_pipeline.py

```diff
@@ -11,88 +11,136 @@
 """
 import numpy as np
 
-from spikeinterface.core import get_chunk_with_margin, get_channel_distances
+from spikeinterface.core import get_chunk_with_margin
 from spikeinterface.core.job_tools import ChunkRecordingExecutor, fix_job_kwargs, _shared_job_kwargs_doc
 
-
+#TODO remove this
 class PeakPipelineStep:
-    """
-    A PeakPipelineStep does a computation on local traces, peaks and optionaly waveforms.
-    It must return an array with shape[0] == peaks.shape[0]
+    pass
 
-    It can be used to compute on-the-fly and in parralel:
-       * peak location
-       * pca (with pretrain model)
-       * some features : ptp, ...
+class PipelineNode:
     """
-    need_waveforms = False
-
-    def __init__(self, recording, ms_before=None, ms_after=None, local_radius_um=None):
-        self._kwargs = dict()
-
-        if self.need_waveforms:
-            assert ms_before is not None
-            assert ms_after is not None
-
-        self.nbefore = None
-        self.nafter = None
-
-        if ms_before is not None:
-            self.nbefore = int(
-                ms_before * recording.get_sampling_frequency() / 1000.)
-            self._kwargs['ms_before'] = float(ms_before)
-
-        if ms_after is not None:
-            self.nafter = int(
-                ms_after * recording.get_sampling_frequency() / 1000.)
-            self._kwargs['ms_after'] = float(ms_after)
-
-        if local_radius_um is not None:
-            # some steps need sparsity mask
-            self._kwargs['local_radius_um'] = float(local_radius_um)
-            self.local_radius_um = local_radius_um
-            self.contact_locations = recording.get_channel_locations()
-            self.channel_distance = get_channel_distances(recording)
-            self.neighbours_mask = self.channel_distance < local_radius_um
+    This is a generic object that will make some computation on peak given a buffer
+    of traces.
+    Typically used for extrating features (amplitudes, localization, ...)
+    
+    A Node can optionally connect to other nodes with the parents and receive inputs from others.
+    """
+    def __init__(self, recording, name, have_global_output, parents=None):
```

Comment:
Suggestion:
I think that `return_ouput` is more clear than `have_global_output`
Maybe there is something better...

Reply:
yep. More elegant. I will do that.

Reply:
Done